### PR TITLE
Update aspera-client-docker image to use the new download workflow

### PR DESCRIPTION
**Pull request by sbesson**
See https://github.com/IDR/idr.openmicroscopy.org/issues/190

This updates the source of the https://hub.docker.com/r/imagedata/download Docker image to use the new Aspera download workflow. This assumes the IDR team wants to keep supporting this image and download workflow. Otherwise, it should probably be archived/deprecated

Included in this PR are the minimal change to make the command functional again

- Drops the study input in favor of the generic fasp-public user
- Update the README with the new command format
- Update the GitHub workflow to use recent GitHub actions and Docker labels
- Upgrade the base image  to Rocky Linux  9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,33 +4,11 @@ IDR Download
...
-| idr0028 | 20160818-original/Images/LM2_GEFGAP_ONTARGETPlus/LM2_GEFGAP_ONTARGETPlus_1A                                 | LM2_ONTARGETPlus_1A                         | 12G  |
-| idr0028 | 20160818-original/Images/LM2_GEFGAP_siGENOME /LM2_GEFGAP_siGENOME_1A                                        | LM2_siGENOME_1A                             | 7.3G |
-| idr0028 | 20160818-original/Images/MDA-MB-231_GEFGAP_ONTARGETPlus/MDA-MB-231_GEFGAP_ONTARGETPlus_1A                   | MDA-MB-231_ONTARGETPlus_1A                  | 11G  |
-| idr0030 | 20170110-original/200972726[5176]_kinome_YAP_TAZ/MeasurementIndex.ColumbusIDX.xml                           | 200972726[5176]_kinome_YAP_TAZ              | 13G  |
-| idr0034 | 20170421-original/Nathalie/HipSci Experiment 10 Plate 1__2014-08-21T17_57_35-Measurement1                   | experiment_10                               | 5.1G |
+        docker run -ti --rm -v /tmp/data:/data imagedata/download SOURCE TARGET
+
+where SOURCE can be any subfolder under https://ftp.ebi.ac.uk/pub/databases/IDR/. For instance
```
Comment (joshmoore) [inline README.md:9]: :heart:

**Comment by pwalczysko** [comment]
Test 1 on Mac M1 - build and run the docker as-is in this PR, use the ``docker run...`` cmd from README and building an image ``test-seb``:

```
aspera-client-docker %  docker build -f Dockerfile --rm --no-cache -t test-seb . 
aspera-client-docker %  docker run -ti --rm -v /Users/pwalczysko:/data test-seb idr0001-graml-sysgro/20151116-verified/JL_120731_S6A /data/JL_120731_S6A 
/home/data/.aspera/connect/bin/ascp -P33001 -l 40m -i /home/data/.aspera/connect/etc/asperaweb_id_dsa.openssh -k 0 fasp-public@fasp.ebi.ac.uk:/pub/databases/IDR/idr0001-graml-sysgro/20151116-verified/JL_120731_S6A /data/JL_120731_S6A
qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory
```

As per https://stackoverflow.com/questions/71040681/qemu-x86-64-could-not-open-lib64-ld-linux-x86-64-so-2-no-such-file-or-direc/71611002#71611002, this ^^^ is caused by me being on M1. Went for a solution where I add the line into the Dockerfile ``FROM --platform=linux/amd64 centos:centos7`` as per line 3 in https://github.com/sbesson/aspera-client-docker/pull/1/commits/ab405c32ecb7a4157952aa595f9a0cec75a40e6f, saved the file as DockerfileM1 and build a new image ``test-sebm1``.  This works fine, see below.


```
aspera-client-docker % docker build -f DockerfileM1 --rm --no-cache -t test-sebm1 .
aspera-client-docker % docker run -ti --rm -v /Users/pwalczysko:/data test-sebm1 idr0001-graml-sysgro/20151116-verified/JL_120731_S6A /data/JL_120731_S6A
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
/home/data/.aspera/connect/bin/ascp -P33001 -l 40m -i /home/data/.aspera/connect/etc/asperaweb_id_dsa.openssh -k 0 fasp-public@fasp.ebi.ac.uk:/pub/databases/IDR/idr0001-graml-sysgro/20151116-verified/JL_120731_S6A /data/JL_120731_S6A
001005001.flex                                                     53%   47MB 38.9Mb/s    00:09 ETA
```

Note: Adding a ``--platform`` flag to the ``docker run...`` cmd was not successful in my hands. 
e.g.
```
docker run -ti --rm -v --platform linux/amd64 /Users/pwalczysko:/data test-seb2 idr0001-graml-sysgro/20151116-verified/JL_120731_S6A /data/JL_120731_S6A
Unable to find image 'linux/amd64:latest' locally
docker: Error response from daemon: pull access denied for linux/amd64, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
```


Only the Dockerfile edit worked.

**Comment by pwalczysko** [comment]
Is it reasonable to keep using centos7 images for this workflow ?

**Review by pwalczysko — CHANGES_REQUESTED** [review]
Two issues:
1. Please comment on the M1 support - imho, this type of platform is more and more prevalent and we could offer a solution in this repo.
2. centos7 ?

**Comment by sbesson** [comment]
Last commit should allow to build the Linux/amd64 image independently on whether the build machine is `arm64` or `amd64`. Thanks @pwalczysko 

On the base image, I am of the opinion we should change it to either Ubuntu 22.04 or Rocky Linux 9 but am not opinionated on either of them. Happy to follow the guidance of the IDR team

**Comment by jburel** [comment]
since the initial image was centOS 7, it might be more natural to switch to Rocky Linux 9

**Comment by pwalczysko** [comment]
Tested last 2 commits (Rocky Linux 9 based image).

Works fine on 

   - Mac Apple M1 Pro
   - Mac amd64 (old laptop)

**Review by pwalczysko — APPROVED** [review]

**Review by jburel — APPROVED** [review]

**Review by joshmoore — APPROVED** [review]
:+1: (Here's to an `AND` vs `OR` option for the requested reviewers :wink:)

**Comment by sbesson** [comment]
Unless someone wants to take the lead on this, I'll merge this PR by EOB and tag 0.4.0, hoping the deployment logic is still functional

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -11,31 +11,28 @@ jobs:
...
-            echo "::set-output name=prefix::${{ github.repository }}:"
-          fi
       - name: Get other tags
         id: gettags
-        uses: jupyterhub/action-major-minor-tag-calculator@v1.1.0
+        uses: jupyterhub/action-major-minor-tag-calculator@v3
         with:
           githubToken: ${{ secrets.GITHUB_TOKEN }}
-          prefix: "${{ steps.getprefix.outputs.prefix }}"
+          prefix: "${{ env.name }}"
```
Comment (sbesson) [inline .github/workflows/main.yml:30]: Looks like I missed there was a breaking change and a `:` need to be appended as in https://github.com/ome/omero-server-docker/commit/6f317379d5a5d191200a491281862a334bd349c3.

Unfortunately, this has created quite a few unwanted images at https://hub.docker.com/u/imagedata. @jburel @joshmoore do you have the credentials to delete those? I can open a PR for a follow-up 0.4.1 release (unless you'd prefer me to delete the tag).

**Pull request merged**